dart-impl/mpi/src/dart_synchronization.c

```diff
@@ -23,7 +23,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <malloc.h>
+#ifdef __APPLE__
+  #include <malloc/malloc.h>
+#else
+  #include <malloc.h>
```

Comment:
Do we really need `malloc.h` or can it be removed alltoghether? We already include `stdlib.h` (which provides `malloc`) and I don't see any other functions being used...

Reply:
I just respect the original include :) if there is no use of `malloc.h`, it is safe to remove